lib/omnibus/packagers/deb.rb

```diff
@@ -390,6 +415,73 @@ def compression_params
     end
 
     #
+    # Check for the existance of a command in +ENV['PATH']+. Exit code +127+
+    # tells us the shell could not find the program.
+    #
+    # @return [bool]
+    #
+    def check_command_exists(command)
+      !(shellout("#{command} </dev/null >/dev/null 2>&1").exitstatus == 127)
+    end
+
+    #
+    # Sign the  +.deb+ file with gpg. This has to be done as separate steps
+    # from creating the +.deb+ file. See +debsigs+ source for behavior
+    # replicated here. +https://gitlab.com/debsigs/debsigs/blob/master/debsigs.txt#L103-124+
+    #
+    # @return [void]
+    def sign_deb_file
+      if signing_passphrase
+        log.info(log_key) { "Signing enabled for .deb file" }
+
+        # Check our dependencies and determine command for GnuPG
+        gpg = nil
+        if check_command_exists('gpg2')
+          gpg = 'gpg2'
+        elsif check_command_exists('gpg')
+          gpg = 'gpg'
+        end
+
+        if gpg && check_command_exists('ar')
+          # Create a directory that will be cleaned when we leave the block
+          Dir.mktmpdir { |tmp_dir|
+            Dir.chdir(tmp_dir) do
+              # Extract the deb file contents
+              shellout!("ar x #{Config.package_dir}/#{package_name}")
+              # Concantenate contents, in order per +debsigs+ documentation.
+              shellout!("cat debian-binary control.tar.* data.tar.* > complete")
+              # Create signature (as +root+)
+              gpg_command =  "#{gpg} --armor --sign --detach-sign"
+              gpg_command << " --local-user '#{project.maintainer}'"
+              gpg_command << " --homedir #{ENV['HOME']}/.gnupg" # TODO: Make this configurable
```

Comment:
Agreed, it should be a configuration option, or at least an environment variable.

Reply:
As of now, it lines up with the RPM packager: lib/omnibus/packagers/rpm.rb:438

Reply:
Oh I see, OK.